internal/middleware/enforce_identity.go

```diff
@@ -0,0 +1,89 @@
+package middleware
```

Comment:
Can you add a comment where this code comes from and why we moved it here please?